docker/mcp/app.py

```
@@ -0,0 +1,94 @@
+        # Return response
+        return SearchResponse(
+            query=query,
+            result_count=len(processed_results),
+            results=processed_results
+        )
+        
+    except HTTPException:
+        raise
+    except Exception as e:
+        logging.error(f"Search error: {str(e)}")
+        raise HTTPException(status_code=500, detail=f"Error: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)